src/api/auth.py

```diff
@@ -28,17 +28,17 @@
         raise RuntimeError("Unable to retrieve API KEY, please ensure the secret ARN is correct")
     except KeyError:
         raise RuntimeError('Please ensure the secret contains a "api_key" field')
-elif api_key_env:
+elif api_key_env != None:
     api_key = api_key_env
 else:
     # For local use only.
     api_key = DEFAULT_API_KEYS
 
-security = HTTPBearer()
+security = HTTPBearer(auto_error=api_key != "")
```

Comment:
I'm nervous about requiring `OPENAI_API_KEY=""` in order to ignore auth. Could we instead just use it if it's there, otherwise, allow it to be missing?

I think this would do it, but I might be wrong.

```suggestion
security = HTTPBearer(auto_error=api_key != None)
```

Reply:
This would break compatibility with upstream though.

Reply:
I agree with you.

Reply:
> This would break compatibility with upstream though.
I don't think we need to care about this

Reply:
We'll stick to `""` as an explicit signal to turn auth off.